### PR TITLE
[release/8.0] Manually update release/8.0 arcade to latest

### PR DESCRIPTION
**Pull request by lewing**
Arcade flow is stuck somewhere in release/8.0 this the result of running `darc update-dependencies --id 194643` manually.

cc @ViktorHofer @mmitche

### CODE REVIEW

**Comment by lewing** [comment]
cc @ericstj

**Comment by lewing** [comment]
@elinor-fung are the host failures here rid related?

File: global.json [hunk 1]
```diff
@@ -1,16 +1,16 @@
 {
   "sdk": {
-    "version": "8.0.100-preview.7.23376.3",
+    "version": "8.0.100-rc.1.23455.8",
```
Comment (carlossanlop) [inline global.json:3]: @lewing Should this be rtm? Also the one below.

Comment (carlossanlop) [reply global.json:3]: Just asking because we've been doing similar manual adjustments in wasm flows. I don't know if it applies to this as well.

Comment (ericstj) [reply global.json:3]: Not yet, we have to wait until we have an RTM SDK.  I believe the arcade insertion is just doing it's thing propagating the arcade toolset changes here.

**Comment by ericstj** [comment]
I noticed we took the RC toolset update in https://github.com/dotnet/runtime/pull/90880/commits/02bcb36e8dfcd82732a33025af420c408b667552.   Among those other changes might be things needed here to update the toolset.

**Comment by agocke** [comment]
The problem is the src/tests/Common/external/external.csproj project. It specifies `win7-x64` as a TFM. But it's even stranger because the comment says the RID doesn't even matter.

The whole thing smells of giant hacks, but I bet updating to `win-x64` will fix the problem, even though I don't really know what's going on.

**Comment by elinor-fung** [comment]
> The whole thing smells of giant hacks, but I bet updating to win-x64 will fix the problem, even though I don't really know what's going on.

This - `win-x86'/`win-x64` - is what we have in main. (I'm also unclear as to what it is doing or what that comment means.)

File: global.json (hunk 1 above)
Comment (lewing) [reply global.json:3]: We'll need a newer sdk to be able to take the new roslyn in https://github.com/dotnet/runtime/pull/92503 but this pr is just to catch us up with normal arcade flow

**Comment by ericstj** [comment]
We don't have to do it in this PR, but I bet we can remove that RID.   It looks to me like this dates back to 1.1 days where we had implementations in RID-specific packages.  So in order to get NuGet to restore the right stuff to copy a RID that plugged into the graph was needed. https://github.com/dotnet/runtime/commit/b843687b2540c55d141e6a1d9b60c86867efad08#diff-61c4029d873ebe0ce8a16d30a697127a7c96e11a1cc018936df1963aba0ef40bR12

**Comment by ViktorHofer** [comment]
> The problem is the src/tests/Common/external/external.csproj project. It specifies win7-x64 as a TFM. But it's even stranger because the comment says the RID doesn't even matter.
The whole thing smells of giant hacks, but I bet updating to win-x64 will fix the problem, even though I don't really know what's going on.

AFAIK, after the runtime tests are all migrated to the new source generated model (which should be completed soon), they will participate in NuGet restore and will get their dependencies directly, instead of via the external.csproj project. That project and  setting a RID for restore will then go away.

**Comment by lewing** [comment]
So installer rid resolution failures left (other than CI noise)?

**Comment by ViktorHofer** [comment]
Need to be addressed here. Those are real failures.

**Comment by elinor-fung** [comment]
`_RequiresILLinkPack` ended up being true for packaging projects based on the single-file analyzer being enabled.  As a result, this was trying to process framework references (it didn't before), which ended up adding those packages as download dependencies. https://github.com/dotnet/sdk/blob/4a1ffab37158144f8b1bc5995241b7ace7aa19ea/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets#L63

Last commit ports over the below from main:
https://github.com/dotnet/runtime/blob/9dfbb61a50269633580f47ac8488a7b10a91911d/eng/Analyzers.targets#L2-L8

**Review by carlossanlop — APPROVED** [review]
CI is green and the two remaining feedback items were addressed by @elinor-fung and @lewing. Merging now.

**Comment by lewing** [comment]
> `_RequiresILLinkPack` ended up being true for packaging projects based on the single-file analyzer being enabled. As a result, this was trying to process framework references (it didn't before), which ended up adding those packages as download dependencies. https://github.com/dotnet/sdk/blob/4a1ffab37158144f8b1bc5995241b7ace7aa19ea/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets#L63
> 
> Last commit ports over the below from main:
> 
> https://github.com/dotnet/runtime/blob/9dfbb61a50269633580f47ac8488a7b10a91911d/eng/Analyzers.targets#L2-L8

Many thanks for resolving that.

**Pull request merged**